sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/SubstationServiceTest.java

```diff
@@ -0,0 +1,133 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.scl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.lfenergy.compas.scl2007b4.model.SCL;
+import org.lfenergy.compas.scl2007b4.model.TLNode;
+import org.lfenergy.compas.scl2007b4.model.TSubstation;
+import org.lfenergy.compas.sct.commons.exception.ScdException;
+import org.lfenergy.compas.sct.commons.testhelpers.SclTestMarshaller;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class SubstationServiceTest {
+
+    @ParameterizedTest
+    @ValueSource(strings = {"/scd-substation-import-ssd/ssd_with_2_substations.xml", "/scd-substation-import-ssd/ssd_without_substations.xml"})
+    void testAddSubstation_Check_SSD_Validity(String ssdFileName) throws Exception {
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scl-root-test-schema-conf/add_ied_test.xml");
+        SCL ssd = SclTestMarshaller.getSCLFromFile(ssdFileName);
+
+        assertThrows(ScdException.class,
+            () -> SubstationService.addSubstation(scd, ssd));
+    }
+
+    @Test
+    void testAddSubstation_SCD_Without_Substation() throws Exception {
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scl-root-test-schema-conf/add_ied_test.xml");
+        SclRootAdapter scdRootAdapter = new SclRootAdapter(scd);
+        SCL ssd = SclTestMarshaller.getSCLFromFile("/scd-substation-import-ssd/ssd.xml");
+        SclRootAdapter ssdRootAdapter = new SclRootAdapter(ssd);
+        SclRootAdapter expectedScdAdapter = SubstationService.addSubstation(scd, ssd);
+
+        assertNotEquals(scdRootAdapter, expectedScdAdapter);
+        assertEquals(expectedScdAdapter.getCurrentElem().getSubstation(), ssdRootAdapter.getCurrentElem().getSubstation());
+    }
+
+    @Test
+    void testAddSubstation_SCD_With_Different_Substation_Name() throws Exception {
```

Comment:
It should throw an exception ? Why ?

Reply:
Because Substation names are different (As in a SCD we can't have more than one Substation).